frontend/www/js/omegaup/components/contest/Clonev2.vue

```diff
@@ -0,0 +1,92 @@
+<template>
+  <div class="card">
+    <div class="card-body">
+      <form
+        class="form"
+        @submit.prevent="
+          $emit('emit-clone', title, alias, description, startTime)
+        "
+      >
+        <div class="row">
+          <div class="form-group col-md-6">
+            <label class="font-weight-bold">{{ T.wordsTitle }}</label>
+            <input v-model="title" class="form-control" size="30" type="text" />
+          </div>
+          <div class="form-group col-md-6">
+            <label class="font-weight-bold">
+              {{ T.contestNewFormShortTitleAlias }}
+              <font-awesome-icon
+                :title="T.contestNewFormShortTitleAliasDesc"
+                icon="info-circle"
+              />
+            </label>
+            <input v-model="alias" class="form-control" type="text" />
+          </div>
+        </div>
+        <div class="row">
+          <div class="form-group col-md-6">
+            <label class="font-weight-bold">{{
+              T.contestNewFormDescription
+            }}</label>
+            <textarea
+              v-model="description"
+              class="form-control"
+              cols="30"
+              rows="10"
+            ></textarea>
+          </div>
+          <div class="form-group col-md-4">
+            <label class="font-weight-bold">
+              {{ T.contestNewFormStartDate }}
+              <font-awesome-icon
+                :title="T.contestNewFormStartDateDesc"
+                icon="info-circle"
+              />
+            </label>
+            <omegaup-datetime v-model="startTime"></omegaup-datetime>
+          </div>
+        </div>
+        <div class="form-group">
+          <button class="btn btn-primary" type="submit">
+            {{ T.wordsCloneContest }}
+          </button>
+        </div>
+      </form>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, Component } from 'vue-property-decorator';
+import T from '../../lang';
+import DateTime from '../DateTimePicker.vue';
+
+import {
+  FontAwesomeIcon,
+  FontAwesomeLayers,
+  FontAwesomeLayersText,
+} from '@fortawesome/vue-fontawesome';
+import { fas } from '@fortawesome/free-solid-svg-icons';
+import { library } from '@fortawesome/fontawesome-svg-core';
+library.add(fas);
+
+@Component({
+  components: {
+    'omegaup-datetime': DateTime,
+    'font-awesome-icon': FontAwesomeIcon,
+    'font-awesome-layers': FontAwesomeLayers,
+    'font-awesome-layers-text': FontAwesomeLayersText,
+  },
+})
+export default class Clone extends Vue {
+  T = T;
+  title = '';
+  alias = '';
+  description = '';
+  startTime = new Date();
+
+  onSubmit() {
+    this.$emit('emit-clone', this);
```

Comment:
qué pasa si aquí se usa

```suggestion
    this.$emit('clone', { title: this.title, alias: this.alias, description: this.description, startTime: this.startTime });
```

y en Editv2.vue se cambia a

```vue
        <omegaup-contest-clone
          @clone="
            ({ title, alias, description, startTime }) =>
              $emit('clone-contest', title, alias, description, startTime)
          "
        ></omegaup-contest-clone>
```

? si eso funciona, este tipo de destructura es menos propenso a errores porque ahora se pueden pasar los argumentos en cualquier orden! si no funciona, al menos hay que cambiar esta línea a

```suggestion
    this.$emit('clone', this.title, this.alias, this.description, this.startTime);
```

hace poco dejamos de usar el prefijo `emit-` para nombres de eventos porque es redundante. y también estamos intentando evitar el uso de `this` en eventos para que sea todo más fácil de probar.

y ya que estamos en eso, se puede agregar una prueba donde se presione el botón y los argumentos esperados se envíen en el evento?

Reply:
Funcionó lo primero.

Me puedes ayudar con la prueba? Por alguna razón el `.trigger('click')` no está funcionando como debería.

Reply:
@pabo99 halp!

Reply:
Claro, lo reviso en un momento.

Reply:
Encontré la solución acá: https://stackoverflow.com/questions/53382235/trigger-form-submit-on-button-click-in-vue-unit-test

Lo que dicen es que Vue Test Utils no adjunta nodos del DOM al documento por default. Entonces hay que usar el `attachTo` en algún elemento del HTML donde vayas a montar el componente. Siguiendo el ejemplo de stackoverflow, así pudieron pasar las dos pruebas:

```typescript
describe('Clonev2.vue', () => {
  const div = document.createElement('div');
  div.id = 'root';
  document.body.appendChild(div);

  it('Should display the form', async () => {
    const wrapper = shallowMount(contest_Clone);

    expect(wrapper.text()).toContain(T.wordsTitle);
  });

  it('Should pass the right arguments to event', async () => {
    const wrapper = shallowMount(contest_Clone, {
      attachTo: '#root',
    });

    const contest = {
      alias: 'contestAlias',
      title: 'Contest Title',
      description: 'Contest description.',
    };
    await wrapper.setData(contest);

    await wrapper.find('button[type="submit"]').trigger('click');
    expect(wrapper.emitted('clone')).toBeDefined();
  });
```

Reply:
Gracias por encontrar la solución @pabo99 !!